### ISSUE TITLE
Darktable (origin/master) crashes  randomly in lighttable and constantly when entering darkroom 

### ISSUE THREAD

**Issue by bhlevca** [darktable-org/darktable#4952]
**Describe the bug**

After opening the freshly compiled darktable and after converting the database all the pictures and collections from the previous version show up.
When clicking on the Import->folder  the application crashes almost immediately. A backtrace is provided below. 

I tried with a completely new config file (wiped out the existing one to make sure that no configurations issues produce the error. 

It happens every time

**To Reproduce**

**Expected behavior**

Expect not to crash but see the imported pictures being populated in lighttable

**Screenshots**

N/A

**Platform (please complete the following information):**
 - Darktable Version: origin/master
 - OS: OpenSuse Tumbleweed
 - OpenCL activated  --> yes
 - Which graphics card and driver version -->NVIDIA k620,  v440.80
[darktable_bt_T93ZJ0.txt](https://github.com/darktable-org/darktable/files/4600471/darktable_bt_T93ZJ0.txt)

**Additional context**

**Comment by Nilvus** [comment]
Last master is about 90 commits since release you use. On actual master, I have no crash (last import tried just this day). Actual master is 3.1.0+1481 and you use 3.1.0+1393.
You could also have sometimes crash related to your system or even the package you use. darktable team do not provide support for packages like this as they do not create them. And the Opensuse master repository is an automatic build one.

So please test a new master and if possible by compiling it. You could find some help to do that on the Github wiki.

**Comment by bhlevca** [comment]
I have 3.1.0+1480-g1b14fcc4b 

It was pulled 2 hours ago and I compiled it myself. If you open the file I have originally attached you will see the version, it is on the first line

**Comment by Nilvus** [comment]
> I have 3.1.0+1480-g1b14fcc4b
> 
> It was pulled 2 hours ago and I compiled it myself. If you open the file I have originally attached you will see the version, it is on the first line and I wlso specified in the subject that it is origin/master and not an OpenSuse package. Actually that package is 3.0.3 and works just fine.

I have open the attached file and on first line it's written: 

`darktable 3.1.0+1393~g44d7ff02a-dirty`

And there's no 3.0.3 release. Last one is 3.0.2. So maybe there's some specific issues on Opensuse repository. Packagers are on pixl.us forum.

**Comment by bhlevca** [comment]
Yes, unfortunately I loaded an old file  here is the one form today

[darktable_bt_YJDBK0.txt](https://github.com/darktable-org/darktable/files/4600825/darktable_bt_YJDBK0.txt)

No, it is 3.0.2 I just wrote from my memory.

The fact of the matter is that 1480 is still crashing. I picked up a few more changes and I will test with that, perhaps i the 1481 version you mentioned.

**Comment by bhlevca** [comment]
I deleted .config/darktable and tested with this one (+1480) and It imported 5-6 pictures after which I got 

```
...
[exif] dt_exif_xmp_read for /multimedia/Picture_Archive/RAW/2020/Flash-tests/_BHP0361.ARW.xmp, id 57 found auto_presets_applied but there was no history
[exif] dt_exif_xmp_read for /multimedia/Picture_Archive/RAW/2020/Flash-tests/_BHP0361.ARW.xmp, id 57 found auto_presets_applied but there was no history
free(): invalid next size (fast)
Aborted (core dumped)
```

**Comment by GrahamByrnes** [comment]
@bhlevca Have you tried again with a more recent master? I also had a series of crashes with +1481, which disappeared when I moved to 1496. They were different, but were due to changes in the CSS (I'll trust @Nilvus on that... he said it would work, it did). 

Since there were a lot of changes done around that time, it might be worth trying again to see if the problem has been solved.

**Comment by MStraeten** [comment]
can confirm an import issue with current master on osx when reimporting a directory already imported with existing xmp files. After deleting the xmp files a reimport doesn't crash.  
 
```
darktable --version
this is darktable 3.1.0+1511~ge16dde635
copyright (c) 2009-2020 johannes hanika
darktable-dev@lists.darktable.org

compile options:
  bit depth is 64 bit
  normal build
  SSE2 optimized codepath enabled
  OpenMP support enabled
  OpenCL support enabled
  Lua support enabled, API version 5.0.2
  Colord support disabled
  gPhoto2 support enabled
  GraphicsMagick support enabled
  OpenEXR support enabled
```
[darktable_crashreport.txt](https://github.com/darktable-org/darktable/files/4606639/darktable_crashreport.txt)

**Comment by bhlevca** [comment]
Actually it is worse now. It would not open at first crashing while loading.  I deleted the ` ~/.config` file and I could have it started, I imported a directory again and it crashed midway. A treace dump is attached.  I have version `3.1.0+1511~ge16dde635`:

[darktable_bt_WC4AK0.txt](https://github.com/darktable-org/darktable/files/4606688/darktable_bt_WC4AK0.txt)

I am on Linux.

**Comment by jenshannoschwalm** [comment]
I think we need one of the xmp files to see what's wrong with them.

**Comment by bhlevca** [comment]
@jenshannoschwalm You got it right.  You gave me the idea to see what is in those directories.

Here is what I noticed. 

1.  If I delete the config file and the xmp files  ( or a new directory only with raw files) the directory imports correctly. 
2.  If I import  the directory that already had an xmp file from a previous vrersion it segfaults.

Here is a new xmp which is corectly created


 
[_BHP0486.ARW.txt](https://github.com/darktable-org/darktable/files/4606740/_BHP0486.ARW.txt)


Here is one that it was created before and its conversion crashes darktable

[_BHP0395.ARW.txt](https://github.com/darktable-org/darktable/files/4606742/_BHP0395.ARW.txt)

rename them back to xmp, github does not allow attachment of xmp extensions

**Comment by bhlevca** [comment]
I noticed something else with this version. If you have nested directories (see below) and there is no picture file in the immediate branch the import is refused.  This used to work  in previous versions, I think.  If not it should be made so. Should I open another bug report?

For example 

```
/Europe
            /Netherlands
                               /pic1.raw
                               /pic2.raw
            /France
                               /pic1.raw
                               /pic2.raw
            /Belgium
                              /pic1.raw
                              /pic2.raw
```

**Comment by jenshannoschwalm** [comment]
I can't reproduce the issue here. To make sure we don't have a module-parameter-xmp issue, please share the image file for  _BHP0486.ARW.txt as that might be broken.

**Comment by bhlevca** [comment]
The image is not broken, there are many in that directory as you can see from the log file. It works perfectly in v3.0.2. It crashes immediately in master at import time when there are xmp files from a previous version of darktable. And it takes longer to crash when  the importex directory had only raw files.

I compiled darktable many times since version 1. It is the first time I experience something like this.

Let me know if you need something  else, but chasing after corrupt raw files is  not the right direction.

**Comment by jenshannoschwalm** [comment]
I do not doubt the correctness of the raw image at all.

What you describe sounds pretty bad as correctness of the sidecars and dt beeing able to reimport is critical.

 If you would like others to check how the specific xmp file for an image leads to problems we mostly need both files. 

The xmp looks good on first sight, it might well be module parameters found in the xmp are missing or wrong. There were issues like this before when module parameters were changed without a proper version check.

This is just one idea i would follow ...

**Comment by MStraeten** [comment]
my scenario:
starting from scatch: 
- fresh build after `git reset --hard origin/master` and `rm -rf *` in build and install directory 
- delete .config/darktable directory
- delete xmps from test directory to be imported

steps:
1. start darktable
2. import test directory
3. close darktable
4. restart darktable
5. import test directory --> crash
darktable also crashes when reloading the xmp via "load sidecar file ..."

so this affect each xmp even it has been created with the same darktable build.

**Comment by MStraeten** [comment]
lldb output:
```
Process 64741 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_INSTRUCTION (code=EXC_I386_INVOP, subcode=0x0)
    frame #0: 0x00000001001653b3 libdarktable.dylib`dt_exif_xmp_read + 16659
libdarktable.dylib`dt_exif_xmp_read:
->  0x1001653b3 <+16659>: ud2
    0x1001653b5 <+16661>: movq   0x2c9d5c(%rip), %rax      ; (void *)0x00007fff98032660: __stderrp
    0x1001653bc <+16668>: movq   %rax, %r14
    0x1001653bf <+16671>: movq   (%rax), %rdi
Target 0: (darktable) stopped.
```

**Comment by jenshannoschwalm** [comment]
Did that here too and could not reproduce. And your log shows a crash while reading xmp. But what presets are used?

So what's wrong with asking for a pair of files? Or maybe you already know the problem...

**Comment by MStraeten** [comment]
just did a new git pull --recurse-submodules https://github.com/darktable-org/darktable.git master --> no longer reproducible 
sample file
https://mega.nz/file/ffpmRCzR#bUE1Ti4Wbx_ikwZZCwcFUbXSClvW8yZUTl6vt74qK7I
and the former crashing xmp
[20190223-IMG_5937.CR2.xmp.txt](https://github.com/darktable-org/darktable/files/4610129/20190223-IMG_5937.CR2.xmp.txt)

**Comment by bhlevca** [comment]
The import works now just fine in +1511, however by changing from one collection to another after a few such operations it still crashes, See the output below. 3.0.2 does not do that on operations with the same data.

Please let me know what and if you need anything from me to find the problem. 

```
/opt/darktable/bin/darktable[dt_exif_xmp_write] /multimedia/Picture_Archive/RAW/2018/Portugal/10380606/_DSC3981.ARW.xmp: caught exiv2 exception '[xmp_write] failed to serialize xmp data'
[dt_exif_xmp_write] /multimedia/Picture_Archive/RAW/2018/Portugal/10080523/_DSC3229.ARW.xmp: caught exiv2 exception '[xmp_write] failed to serialize xmp data'
[dt_exif_xmp_write] /multimedia/Picture_Archive/RAW/2018/Portugal/10280522/DSC03082.ARW.xmp: caught exiv2 exception '[xmp_write] failed to serialize xmp data'
[dt_exif_xmp_write] /multimedia/Picture_Archive/RAW/2018/Portugal/10080602/_DSC3928.ARW.xmp: caught exiv2 exception '[xmp_write] failed to serialize xmp data'
[dt_exif_xmp_write] /multimedia/Picture_Archive/RAW/2018/Portugal/10980601/_DSC3889.ARW.xmp: caught exiv2 exception '[xmp_write] failed to serialize xmp data'
[_ioppr_check_rules] found rule flip clipping module clipping (17) is before flip (21) image 925 (dt_dev_read_history_no_image end)
[_ioppr_check_rules] found rule flip clipping module flip (21) is after clipping (17) image 925 (dt_dev_read_history_no_image end)
darktable: malloc.c:4036: _int_malloc: Assertion `(unsigned long) (size) >= (unsigned long) (nb)' failed.
Aborted (core dumped)
```

**Comment by bhlevca** [comment]
I found a way to crash it consistently. 

1. open darktable (obviusly light table mode)
2. doubleclick on a  picture (enter darkroom)
3. crash occurs all the time with the following output:
```
/opt/darktable/bin/darktable
darktable: malloc.c:2379: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
Aborted (core dumped)
```

**Comment by ptilopteri** [comment]
@bhlevca fwiw, I downloaded your example image and xmp file and imported into 
darktable-3.1.0~git1540.912ea6b00 w/o a problem, double clicked on image thumbnail in lighttable and it opened in darkroom.  No crash/seg-fault

openSUSE Tumbleweed 20200509
NVIDIA GF106 [GeForce GTS 450], 390.132
darktable-3.1.0~git1540.912ea6b00-2979.1.x86_64
OpenCL on 
i7 12-core 36GB

**Comment by ptilopteri** [comment]
@bhlevca maybe you have a hardware problem.

**Comment by bhlevca** [comment]
> @bhlevca maybe you have a hardware problem.

@ptilopteri You did not read that 3.0.2 works perfectly on the SAME hardware?
No, I don't think that it is a hardware issue it is software, but we need to understand if it is bad code or bad build. I 've been buiding darktable since version 1, Never had a problem. What is new in master that that was not used in 3.0.2

I don't have the time now, but I could run it under the debugger and see where it is segfaulting.  
I have an Nvidia card with proprietary driver 440.82 on a Quadro 620k,  64 GB RAM, OpenCL on

**Comment by jenshannoschwalm** [comment]
There are a number of issues around related to the post 3.0.2 code, especially the new lighttable and culling stuff. Or presets, or ... What you describe could be related to pretty much anything :-)

Unfortunately i can't reproduce here, i guess you already made sure the submodules including rawspeed and OpenCL are updated and you tested with cl on/off.

At least, i guess you should think about this issue. The title is simply misleading and the first parts of the discussion did not correctly describe what's wrong. So you might close this and open another one after

**Comment by bhlevca** [comment]
> There are a number of issues around related to the post 3.0.2 code, especially the new lighttable and culling stuff. Or presets, or ... What you describe could be related to pretty much anything :-)

So there are changes and bugs could have been introduced :-) I suppose you meant "past" 3.0.2 code. Maybe it had issues but it did not crash here. 

> 
> Unfortunately i can't reproduce here, i guess you already made sure the submodules including rawspeed and OpenCL are updated and you tested with cl on/off. I will try at some point today

Yes, I updated every time the submodules. No, I did not turn off Open CL

> At least, i guess you should think about this issue. 
I did I will try without OpenCL, but I was hoping that the trace would be useful for catching the problem.

> The title is simply misleading and the first parts of the discussion did not correctly describe what's wrong. So you might close this and open another one after

This is not quite true. Initially I was not able to import anything, it was crashing at imports. Lots of commits have been done since this was open.  I can simply change the title instead opening a new one.

**Comment by bhlevca** [comment]
I think that I found the problem. I had a few old libs in /usr/local/lbi64, namely lensfun and gmic.
while the headers where taken from a newer library from /usr/include. 

I removed those and rebuilt the app and  tested thoroughly and it does not crash in any of my previous combination reported. The bad thing is that an ldd did not show me anything picked form the /usr/local or I did not look carefully. 

In any case, I hope that I did not waste anyone's time. 

Should I close the bug or it has to be done by the development team?
